### PR TITLE
[docs] Update RLlib and Tune use case example highlights

### PR DESCRIPTION
**Pull request by ericl**
- Add RLlib course and suggestions from @sven1977 
- Update Tune section per suggestions from Kai
- Add gallery icon and use it consistently across sections

### CODE REVIEW

**Review by maxpumperla — APPROVED** [review]

File: doc/source/ray-overview/use-cases.rst [hunk 1]
```diff
@@ -188,44 +188,44 @@ Learn more about the Tune library with the following talks and user guides.
...
         :classes: btn-link btn-block stretched-link
     ---
     :img-top: /images/tune.png
 
     +++
-    .. link-button:: https://druce.ai/2020/10/hyperparameter-tuning-with-xgboost-ray-tune-hyperopt-and-optuna
-        :type: url
-        :text: [Blog] Hyperopt, Optuna, and Ray Tune for XGBoost and LightGBM
+    .. link-button:: https://www.anyscale.com/blog/how-to-distribute-hyperparameter-tuning-using-ray-tune
+        :type: ref
```
Comment (krfricke) [inline doc/source/ray-overview/use-cases.rst:200]:
```suggestion
        :type: url
```
Link does not work without this

**Review by krfricke — APPROVED** [review]
Order is great, thanks!

**Comment by pcmoritz** [comment]
Very nice :)

**Comment by pcmoritz** [comment]
Similar to the discussion in https://github.com/ray-project/ray/pull/30846, let's cherry pick this onto Ray 2.2 (no need to change the release process / wheels, this is only for https://docs.ray.io/en/latest/)

**Pull request merged**